diff --git a/constructor/nsis/main.nsi.tmpl b/constructor/nsis/main.nsi.tmpl
index 6f5c8ae..3a78cd0 100644
--- a/constructor/nsis/main.nsi.tmpl
+++ b/constructor/nsis/main.nsi.tmpl
@@ -37,6 +37,11 @@ Unicode "true"
 !insertmacro GetParameters
 !insertmacro GetOptions
 
+# All commands used from StrFunc.nsh need to be called before they can be used.
+# https://nsis.sourceforge.io/StrFunc#3.1_How_To_Use_Commands_In_Install_Sections_and_Functions
+!include "StrFunc.nsh"
+${StrCase}
+
 !include "UAC.nsh"
 !include "nsDialogs.nsh"
 
@@ -827,7 +832,9 @@ Function OnDirectoryLeave
     Call CheckForSpaces
     Pop $R0 # The function returns the number of spaces found in the input string.
 
-#if check_path_spaces is True
+    Push $R7
+    Push $R8
+    Push $R9
     # Check if any spaces exist in $INSTDIR.
     StrCmp $R0 0 NoSpaces
 
@@ -837,18 +844,35 @@ Function OnDirectoryLeave
         Goto +2
           StrCpy $R1 "s"
 
+        ${If} ${Silent}
+          StrCpy $R7 " "
+        ${Else}
+          StrCpy $R7 "$\n"
+        ${EndIf}
+        StrCpy $R8 "'Destination Folder' contains $R0 space$R1.$R7This can cause problems with several conda packages.$R7"
+#if check_path_spaces is True
+            StrCpy $R8 "$R8Please remove the space$R1 from the destination folder."
+            StrCpy $R9 "error"
+#else
+            StrCpy $R8 "$R8Please consider removing the space$R1."
+            StrCpy $R9 "warning"
+#endif
         # Show message box then take the user back to the Directory page.
-        MessageBox MB_OK|MB_ICONINFORMATION \
-            "Warning: 'Destination Folder' contains $R0 space$R1.$\n \
-             This can cause problems with several Conda packages.$\n \
-             Please consider removing the space$R1." \
-            /SD IDOK
         ${If} ${Silent}
-            DetailPrint "::error:: 'Destination folder' contains $R0 space$R1."
-            abort
+            # The error/warning strings must be lower case so that tests can catch them
+            ${StrCase} $R9 $R9 "L"
+            DetailPrint "::$R9:: $R8"
+        ${Else}
+            ${StrCase} $R9 $R9 "S"
+            MessageBox MB_OK|MB_ICONINFORMATION "$R9: $R8" /SD IDOK
         ${EndIf}
-    NoSpaces:
+#if check_path_spaces is True
+            abort
 #endif
+    NoSpaces:
+    Pop $R7
+    Pop $R8
+    Pop $R9
 
     # List of characters not allowed anywhere in $INSTDIR
     Push "^%!=,"
