diff --git a/constructor/nsis/main.nsi.tmpl b/constructor/nsis/main.nsi.tmpl
index 6f5c8ae..dbe5c25 100644
--- a/constructor/nsis/main.nsi.tmpl
+++ b/constructor/nsis/main.nsi.tmpl
@@ -1111,6 +1111,11 @@ Section "Install"
 
         WriteRegStr SHCTX "${PYREG}\InstallPath" "" "$INSTDIR"
 
+        WriteRegStr $HCTX "${PYREG}\InstallPath" \
+            "ExecutablePath" "$INSTDIR\python.exe"
+        WriteRegStr $HCTX "${PYREG}\InstallPath" \
+            "WindowedExecutablePath" "$INSTDIR\pythonw.exe"
+
         WriteRegStr SHCTX "${PYREG}\InstallPath\InstallGroup" \
             "" "Python ${PY_VER}"
 
