diff --git a/constructor/nsis/main.nsi.tmpl b/constructor/nsis/main.nsi.tmpl
index 6f5c8ae..d6524a7 100644
--- a/constructor/nsis/main.nsi.tmpl
+++ b/constructor/nsis/main.nsi.tmpl
@@ -36,6 +36,7 @@ Unicode "true"
 !include "FileFunc.nsh"
 !insertmacro GetParameters
 !insertmacro GetOptions
+${StrTok}
 
 !include "UAC.nsh"
 !include "nsDialogs.nsh"
@@ -669,6 +670,27 @@ Function .onInit
 FunctionEnd
 
 Function un.onInit
+    # Never run the uninstaller when $INSTDIR points at system-critical directories
+    StrCpy $R0 "ALLUSERSPROFILE APPDATA LOCALAPPDATA PROGRAMDATA PROGRAMFILES PROGRAMFILES(x86) PUBLIC SYSTEMDRIVE SYSTEMROOT USERPROFILE"
+    Push $R0
+    Call CheckForSpaces
+    Pop $R1
+    ${ForEach} $i 0 $R1 + 1
+        ${StrTok} $R2 $R0 " " $i "1"
+        ReadEnvStr $R2
+        StrCmp $R1 $INSTDIR invalid_dir
+    ${Next}
+
+    # Primitive check to see that $INSTDIR points to a conda directory
+    IfFileExists $INSTDIR\_conda.exe valid_conda_dir invalid_dir
+
+    invalid_dir:
+        MessageBox MB_OK|MB_ICONSTOP \
+            "Error: $INSTDIR is Not a valid conda directory. Please run the uninstaller from a conda directory." \
+            /SD IDABORT
+        abort
+    valid_conda_dir:
+
     # Select the correct registry to look at, depending
     # on whether it's a 32-bit or 64-bit installer
     SetRegView @BITS@
@@ -679,15 +701,15 @@ Function un.onInit
     # not automatically elevate the uninstaller for us -- we need to do it
     # ourselves if we're not a 'Just Me' installation.
     !insertmacro UAC_PageElevation_OnInit
-    ${IfNot} ${FileExists} "$INSTDIR\.nonadmin"
+    ${IfNot} ${FileExists} "$INSTDIR\.admin"
     ${AndIfNot} ${UAC_IsAdmin}
         !insertmacro DoElevation
     ${EndIf}
 
-    ${If} ${FileExists} "$INSTDIR\.nonadmin"
-        SetShellVarContext Current
-    ${Else}
+    ${If} ${FileExists} "$INSTDIR\.admin"
         SetShellVarContext All
+    ${Else}
+        SetShellVarContext Current
     ${EndIf}
 FunctionEnd
 
@@ -996,9 +1018,9 @@ Section "Install"
     # Copy extra files (code generated on winexe.py)
     @EXTRA_FILES@
 
-    ${If} $InstMode = ${JUST_ME}
+    ${If} $InstMode = ${ALL_USERS}
         SetOutPath "$INSTDIR"
-        FileOpen $0 ".nonadmin" w
+        FileOpen $0 ".admin" w
         FileClose $0
     ${EndIf}
 
