diff --git a/constructor/nsis/main.nsi.tmpl b/constructor/nsis/main.nsi.tmpl
index 6f5c8ae..420f754 100644
--- a/constructor/nsis/main.nsi.tmpl
+++ b/constructor/nsis/main.nsi.tmpl
@@ -37,6 +37,8 @@ Unicode "true"
 !insertmacro GetParameters
 !insertmacro GetOptions
 
+${UnStrTok}
+
 !include "UAC.nsh"
 !include "nsDialogs.nsh"
 
@@ -669,6 +671,28 @@ Function .onInit
 FunctionEnd
 
 Function un.onInit
+    # Never run the uninstaller when $INSTDIR points at system-critical directories
+    StrCpy $R0 "ALLUSERSPROFILE APPDATA LOCALAPPDATA PROGRAMDATA PROGRAMFILES PROGRAMFILES(x86) PUBLIC SYSTEMDRIVE SYSTEMROOT USERPROFILE"
+    StrCpy $R1 0
+    loop:
+        ${UnStrTok} $R2 $R0 " " $R1 1
+        StrCmp $R2 "" endloop
+        ReadEnvStr $R3 $R2
+        StrCmp $R3 $INSTDIR invalid_dir
+        IntOp $R1 + 1
+        goto loop
+    endloop:
+
+    # Primitive check to see that $INSTDIR points to a conda directory
+    IfFileExists $INSTDIR\_conda.exe valid_conda_dir invalid_dir
+
+    invalid_dir:
+        MessageBox MB_OK|MB_ICONSTOP \
+            "Error: $INSTDIR is Not a valid conda directory. Please run the uninstaller from a conda directory." \
+            /SD IDABORT
+        abort
+    valid_conda_dir:
+
     # Select the correct registry to look at, depending
     # on whether it's a 32-bit or 64-bit installer
     SetRegView @BITS@
