diff --git a/constructor/nsis/main.nsi.tmpl b/constructor/nsis/main.nsi.tmpl
index 6f5c8ae..4f933fd 100644
--- a/constructor/nsis/main.nsi.tmpl
+++ b/constructor/nsis/main.nsi.tmpl
@@ -669,6 +669,42 @@ Function .onInit
 FunctionEnd
 
 Function un.onInit
+    # Never run the uninstaller when $INSTDIR points at system-critical directories
+    ReadEnvStr $R0 ALLUSERSPROFILE
+    StrCmp $R0 $INSTDIR invalid_dir
+    ReadEnvStr $R0 APPDATA
+    StrCmp $R0 $INSTDIR invalid_dir
+    ReadEnvStr $R0 LOCALAPPDATA
+    StrCmp $R0 $INSTDIR invalid_dir
+    ReadEnvStr $R0 PROGRAMDATA
+    StrCmp $R0 $INSTDIR invalid_dir
+    ReadEnvStr $R0 PROGRAMFILES
+    StrCmp $R0 $INSTDIR invalid_dir
+    ReadEnvStr $R0 PROGRAMFILES(x86)
+    StrCmp $R0 $INSTDIR invalid_dir
+    ReadEnvStr $R0 PUBLIC
+    StrCmp $R0 $INSTDIR invalid_dir
+    ReadEnvStr $R0 SYSTEMDRIVE
+    # SYSTEMDRIVE does not contain backslash
+    StrCpy $R0 "$R0\"
+    StrCmp $R0 $INSTDIR invalid_dir
+    ReadEnvStr $R0 SYSTEMROOT
+    StrCmp $R0 $INSTDIR invalid_dir
+    ReadEnvStr $R0 USERPROFILE
+    StrCmp $R0 $INSTDIR invalid_dir
+    ReadEnvStr $R0 WINDIR
+    StrCmp $R0 $INSTDIR invalid_dir
+
+    # Primitive check to see that $INSTDIR points to a conda directory
+    IfFileExists $INSTDIR\_conda.exe valid_conda_dir invalid_dir
+
+    invalid_dir:
+        MessageBox MB_OK|MB_ICONSTOP \
+            "Error: $INSTDIR is Not a valid conda directory. Please run the uninstaller from a conda directory." \
+            /SD IDABORT
+        abort
+    valid_conda_dir:
+
     # Select the correct registry to look at, depending
     # on whether it's a 32-bit or 64-bit installer
     SetRegView @BITS@
@@ -679,15 +715,15 @@ Function un.onInit
     # not automatically elevate the uninstaller for us -- we need to do it
     # ourselves if we're not a 'Just Me' installation.
     !insertmacro UAC_PageElevation_OnInit
-    ${IfNot} ${FileExists} "$INSTDIR\.nonadmin"
+    ${IfNot} ${FileExists} "$INSTDIR\.admin"
     ${AndIfNot} ${UAC_IsAdmin}
         !insertmacro DoElevation
     ${EndIf}
 
-    ${If} ${FileExists} "$INSTDIR\.nonadmin"
-        SetShellVarContext Current
-    ${Else}
+    ${If} ${FileExists} "$INSTDIR\.admin"
         SetShellVarContext All
+    ${Else}
+        SetShellVarContext Current
     ${EndIf}
 FunctionEnd
 
@@ -996,9 +1032,9 @@ Section "Install"
     # Copy extra files (code generated on winexe.py)
     @EXTRA_FILES@
 
-    ${If} $InstMode = ${JUST_ME}
+    ${If} $InstMode = ${ALL_USERS}
         SetOutPath "$INSTDIR"
-        FileOpen $0 ".nonadmin" w
+        FileOpen $0 ".admin" w
         FileClose $0
     ${EndIf}
 
