diff --git a/constructor/nsis/main.nsi.tmpl b/constructor/nsis/main.nsi.tmpl
index 6f5c8ae..704da89 100644
--- a/constructor/nsis/main.nsi.tmpl
+++ b/constructor/nsis/main.nsi.tmpl
@@ -669,6 +669,14 @@ Function .onInit
 FunctionEnd
 
 Function un.onInit
+    IfFileExists $INSTDIR\_conda.exe valid_conda_dir invalid_dir
+    invalid_dir:
+        MessageBox MB_OK|MB_ICONSTOP \
+            "Error: Not a valid conda directory. Please run the uninstaller from a conda directory." \
+            /SD IDABORT
+        abort
+    valid_conda_dir:
+
     # Select the correct registry to look at, depending
     # on whether it's a 32-bit or 64-bit installer
     SetRegView @BITS@
@@ -679,15 +687,15 @@ Function un.onInit
     # not automatically elevate the uninstaller for us -- we need to do it
     # ourselves if we're not a 'Just Me' installation.
     !insertmacro UAC_PageElevation_OnInit
-    ${IfNot} ${FileExists} "$INSTDIR\.nonadmin"
+    ${IfNot} ${FileExists} "$INSTDIR\.admin"
     ${AndIfNot} ${UAC_IsAdmin}
         !insertmacro DoElevation
     ${EndIf}
 
-    ${If} ${FileExists} "$INSTDIR\.nonadmin"
-        SetShellVarContext Current
-    ${Else}
+    ${If} ${FileExists} "$INSTDIR\.admin"
         SetShellVarContext All
+    ${Else}
+        SetShellVarContext Current
     ${EndIf}
 FunctionEnd
 
@@ -996,9 +1004,9 @@ Section "Install"
     # Copy extra files (code generated on winexe.py)
     @EXTRA_FILES@
 
-    ${If} $InstMode = ${JUST_ME}
+    ${If} $InstMode = ${ALL_USERS}
         SetOutPath "$INSTDIR"
-        FileOpen $0 ".nonadmin" w
+        FileOpen $0 ".admin" w
         FileClose $0
     ${EndIf}
 
@@ -1171,7 +1179,6 @@ SectionEnd
 
 Section "Uninstall"
     # Remove menu items, path entries
-
     DetailPrint "Deleting @NAME@ menus..."
     nsExec::ExecToLog '"$INSTDIR\_conda.exe" constructor --prefix "$INSTDIR" --rm-menus'
 
