diff --git a/constructor/header.sh b/constructor/header.sh
index 129d3da..1d4d33c 100644
--- a/constructor/header.sh
+++ b/constructor/header.sh
@@ -249,6 +249,21 @@ then
 #endif
 
 #if aarch64
+    if [ -f /proc/device-tree/model && -n "$(cat /proc/device-tree/model | grep Raspberry)" ]; then
+        printf "WARNING:\\n"
+        printf "    Your system appears to be a RaspberryPi.\\n"
+        printf "    Anaconda packages may not be compatible with this system.\\n"
+        printf "    Are sure you want to continue the installation? [yes|no]\\n"
+        printf "[no] >>> "
+        read -r ans
+        ans=$(echo "${ans}" | tr '[:lower:]' '[:upper:]')
+        if [ "$ans" != "YES" ] && [ "$ans" != "Y" ]
+        then
+            printf "Aborting installation\\n"
+            exit 2
+        fi
+    fi
+
     if [ "$(uname -m)" != "aarch64" ]; then
         printf "WARNING:\\n"
         printf "    Your machine hardware does not appear to be aarch64, \\n"
