diff --git a/tests/test_examples.py b/tests/test_examples.py
index 4bd862a..5721def 100644
--- a/tests/test_examples.py
+++ b/tests/test_examples.py
@@ -349,7 +349,7 @@ def test_example_noconda(tmp_path, request):
         _run_installer(input_path, installer, install_dir, request=request)
 
 
-@pytest.mark.skipif(sys.platform != "Darwin", reason="macOS only")
+@pytest.mark.skipif(sys.platform != "darwin", reason="macOS only")
 def test_example_osxpkg(tmp_path, request):
     input_path = _example_path("osxpkg")
     for installer, install_dir in create_installer(input_path, tmp_path):
